hudi-common/src/test/java/org/apache/hudi/common/schema/TestHoodieSchemaUtils.java

```diff
@@ -0,0 +1,350 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.common.schema;
+
+import org.apache.hudi.avro.AvroSchemaUtils;
+import org.apache.hudi.avro.HoodieAvroUtils;
+
+import org.apache.avro.Schema;
+import org.junit.jupiter.api.Test;
+
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * Tests for {@link HoodieSchemaUtils}.
+ */
+public class TestHoodieSchemaUtils {
+
+  private static final String SIMPLE_SCHEMA = "{"
+      + "\"type\":\"record\","
+      + "\"name\":\"TestRecord\","
+      + "\"fields\":["
+      + "{\"name\":\"id\",\"type\":\"string\"},"
+      + "{\"name\":\"name\",\"type\":\"string\"}"
+      + "]}";
+
+  private static final String EVOLVED_SCHEMA = "{"
+      + "\"type\":\"record\","
+      + "\"name\":\"TestRecord\","
+      + "\"fields\":["
+      + "{\"name\":\"id\",\"type\":\"string\"},"
+      + "{\"name\":\"name\",\"type\":\"string\"},"
+      + "{\"name\":\"age\",\"type\":[\"null\",\"int\"],\"default\":null}"
+      + "]}";
+
+  @Test
+  public void testCreateHoodieWriteSchema() {
+    // Test with operation field
+    HoodieSchema writeSchemaWithOp = HoodieSchemaUtils.createHoodieWriteSchema(SIMPLE_SCHEMA, true);
+
+    assertNotNull(writeSchemaWithOp);
+    assertEquals(HoodieSchemaType.RECORD, writeSchemaWithOp.getType());
+
+    // Should have original fields plus metadata fields
+    List<HoodieSchemaField> fields = writeSchemaWithOp.getFields();
+    assertTrue(fields.size() > 2); // Original 2 fields + metadata fields
+
+    // Test without operation field
+    HoodieSchema writeSchemaNoOp = HoodieSchemaUtils.createHoodieWriteSchema(SIMPLE_SCHEMA, false);
+
+    assertNotNull(writeSchemaNoOp);
+    assertEquals(HoodieSchemaType.RECORD, writeSchemaNoOp.getType());
+
+    // Should have different number of fields
+    assertNotNull(writeSchemaNoOp.getFields());
```

Comment:
This assertion is just check if it is non null, not the number of fields. Should this be updated?

Reply:
I've updated this as well